client/app/intake/pages/review.jsx

```
@@ -55,9 +57,11 @@ class Review extends React.PureComponent {
 
 class ReviewNextButton extends React.PureComponent {
   handleClick = () => {
-    this.props.submitReview(this.props.rampElection).then(
-      () => this.props.history.push('/finish')
-    );
+    this.props.submitReview(this.props.rampElection).then(() => {
+      if (this.props.requestState === REQUEST_STATE.SUCCEEDED) {
```

This should be unnecessary. We're in the success handler for the promise, so the fact that this code is executing should mean that the request succeeded.

---

Ah, I see that we have an error handler in the API call that swallows the error. Just `throw error` at the end of the error handler, and then this conditional will be unnecessary. (Happy to explain more over the phone if it'd help.)